### PR TITLE
Jennyf/claims inner exception 

### PR DESCRIPTION
**Pull request by jennyf19**
#777

### CODE REVIEW

File: src/Microsoft.IdentityModel.Clients.ActiveDirectory/Internal/Http/AdalHttpClient.cs [hunk 1]
```diff
@@ -128,8 +128,7 @@ public AdalHttpClient(string uri, CallState callState)
                     if (ex.WebResponse.StatusCode == HttpStatusCode.BadRequest &&
                         tokenResponse.Error == AdalError.InteractionRequired)
                     {
-                        throw new AdalClaimChallengeException(tokenResponse.Error, tokenResponse.ErrorDescription,
-                            tokenResponse.Claims);
+                        throw new AdalClaimChallengeException(tokenResponse.Error, tokenResponse.ErrorDescription, ex.InnerException, tokenResponse.Claims);
```
Comment (kpanwar) [inline src/Microsoft.IdentityModel.Clients.ActiveDirectory/Internal/Http/AdalHttpClient.cs]: passed in exception seems to be incorrect. check what exception object is passed in when it is not claims challenge exception

**Comment by kpanwar** [comment]
we should update the unit test for this

File: src/Microsoft.IdentityModel.Clients.ActiveDirectory/Internal/Http/AdalHttpClient.cs (hunk 1 above)
Comment (jennyf19) [reply src/Microsoft.IdentityModel.Clients.ActiveDirectory/Internal/Http/AdalHttpClient.cs]: Done.

File: src/Microsoft.IdentityModel.Clients.ActiveDirectory/AdalClaimChallengeException.cs [hunk 2]
```diff
@@ -33,17 +35,17 @@ namespace Microsoft.IdentityModel.Clients.ActiveDirectory
     public class AdalClaimChallengeException : AdalServiceException
     {
         /// <summary>
-        ///  Initializes a new instance of the exception class.
+        /// Claims returned from eSTS during token acquisition.
```
Comment (kpanwar) [inline src/Microsoft.IdentityModel.Clients.ActiveDirectory/AdalClaimChallengeException.cs]: why has the documentation changed?

Comment (jennyf19) [reply src/Microsoft.IdentityModel.Clients.ActiveDirectory/AdalClaimChallengeException.cs]: fixed.

File: src/Microsoft.IdentityModel.Clients.ActiveDirectory/AdalClaimChallengeException.cs [hunk 3]
```diff
@@ -33,17 +35,17 @@ namespace Microsoft.IdentityModel.Clients.ActiveDirectory
     public class AdalClaimChallengeException : AdalServiceException
     {
         /// <summary>
-        ///  Initializes a new instance of the exception class.
+        /// Represents claims returned from AdalClaimChallengeException
```
Comment (kpanwar) [inline src/Microsoft.IdentityModel.Clients.ActiveDirectory/AdalClaimChallengeException.cs]: this is still different

Comment (jennyf19) [reply src/Microsoft.IdentityModel.Clients.ActiveDirectory/AdalClaimChallengeException.cs]: done.

**Comment by kpanwar** [comment]
:shipit:

**Pull request merged**